recipes/libva/all/conanfile.py

```diff
@@ -0,0 +1,144 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import fix_apple_shared_install_name
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, rm, rmdir
+from conan.tools.gnu import PkgConfigDeps
+from conan.tools.layout import basic_layout
+from conan.tools.meson import Meson, MesonToolchain
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class PackageConan(ConanFile):
+    name = "libva"
+    description = "Libva is an implementation for VA-API"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/intel/libva"
+    topics = ("VA-API", "Video", "Acceleration")
+    provides = "vaapi"
+    package_type = "shared-library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "with_drm": [True, False],
+        "with_x11": [True, False],
+        "with_glx": [True, False],
+        "with_wayland": [True, False],
+        "with_win32": [True, False],
+    }
+    default_options = {
+        "with_drm": True,
+        "with_x11": True,
+        "with_glx": True,
+        "with_wayland": True,
+        "with_win32": True,
+    }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os != "Windows":
+            del self.options.with_win32
+
+        if self.settings.os not in ["Linux", "FreeBSD"]:
+            del self.options.with_x11
+            del self.options.with_glx
+            del self.options.with_drm
+
+        if self.settings.os != "Linux":
+            del self.options.with_wayland
+
+    def configure(self):
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        if self.options.get_safe("with_x11"):
+            self.requires("xorg/system")
+        if self.options.get_safe("with_drm"):
+            self.requires("libdrm/2.4.114")
+        if self.options.get_safe("with_wayland"):
+            self.requires("wayland/1.22.0")
+        if self.options.get_safe("with_glx"):
+            self.requires("opengl/system")
+
+    def validate(self):
+        if self.options.get_safe("with_glx") and not self.options.get_safe("with_x11"):
+            raise ConanInvalidConfiguration(f"{self.ref} needs x11 or glx")
```

Comment:
This message could be clearer, I think.

```suggestion
            raise ConanInvalidConfiguration(f"{self.ref} requires x11 when glx is enabled")
```